ts/src/zebpay.ts

```diff
@@ -0,0 +1,1933 @@
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const request: Dict = {
+            'leverage': leverage,
+            'symbol': this.marketId (symbol).toUpperCase (),
+        };
+        //
+        // { data: { "symbol", "longLeverage": 10, "shortLeverage": 1, "marginMode": "isolated" }
+        //
+        params['defaultType'] = 'future';
+        const response = await this.privateFuturePostTradeUpdateUserLeverage (this.extend (request, params));
+        return response;
+    }
+
+    /**
+     * @method
+     * @name zebpay#fetchPositions
+     * @see https://github.com/zebpay/zebpay-api-references/blob/main/futures/api-reference/private-endpoints/trade.md#--get-positions
+     * @description Fetches current contract trading positions
+     * @param {string[]} symbols List of unified symbols
+     * @param {object} [params] Not used by krakenfutures
+     * @returns Parsed exchange response for positions
+     */
+    async fetchPositions (symbols: Strings = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        params['defaultType'] = 'future';
+        const response = await this.privateFutureGetTradePositions (this.extend (request, params));
+        //
+        //    {
+        //        "data": [
+        //            {
+        //                "id": "31998678-6056-413f-9d0d-fc3678641650",
+        //                "symbol": "ETHINR",
+        //                "entryPrice": "0.7533",
+        //                "datetime": "2022-03-03T22:51:16.566Z",
+        //                "contractSize": "230"
+        //            }
+        //        ],
+        //    }
+        //
+        const positions = this.safeList (response, 'data', []);
+        const result = this.parsePositions (positions);
+        return this.filterByArrayPositions (result, 'symbol', symbols, false);
+    }
+
+    /**
+     * @method
+     * @name zebpayfutures#addMargin
+     * @description add margin
+     * @see https://github.com/zebpay/zebpay-api-references/blob/main/futures/api-reference/private-endpoints/trade.md#-add-margin-to-position
+     * @param {string} symbol unified market symbol
+     * @param {float} amount amount of margin to add
+     * @param {object} [params] extra parameters specific to the exchange API endpoint.
+     * @param {string} [params.positionId] PositionId of the order to add margin.
+     * @param {string} [params.timestamp] Tiemstamp.
+     * @returns {object} a [margin structure]{@link https://docs.ccxt.com/#/?id=add-margin-structure}
+     */
+    async addMargin (symbol: string, amount: number, params = {}): Promise<MarginModification> {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const positionId = this.safeString (params, 'positionId');
```

Comment:
you don't need to remove the `positionid` from `params` to later add it back with `this.extend(request, params)` if `positioNid` is already in it, you don't need to do anything else.